textbook/2_grouping.qmd

```diff
@@ -0,0 +1,328 @@
+    y=alt.Y('percentage_heavy_drinkers'),
+    tooltip=['country', 'average_life_expectancy', 'percentage_heavy_drinkers']
+)
+```
+
+
+
+Notice that the data points form clusters by continent?
+
+While countries in both <span style="color: #E37F7E; font-weight: bold">Asia</span> and <span style="color: #9CCCC8; font-weight: bold">Europe</span> tend to have high life expectancies, heavy drinking is more common in <span style="color: #9CCCC8; font-weight: bold">Europe</span>. This makes sense - European countries often have more pronounced drinking cultures. 
+
+Meanwhile, most of the countries with low life expectancies are in <span style="color: #5D84A8; font-weight: bold">Africa</span>. Regional public health problems such as high rates of infant mortality can reduce life expectancy.
+
+By grouping the data, we have revealed that world region influences both life expectancy and alcohol consumption. This is sobering news for Viridia.
+
+:::
+
+::::
+
+:::::
+
+Coloring data points is a powerful example of a technique known as **grouping**. The goal of grouping is to distinguish between groups by basing the visual appearance of data points on the value of categorical variable. 
+
+As we have seen with alcohol consumption and life expectancy, grouping has major benefits. It allows us to:
+
+1. **Directly compare** trends and values for distinct groups
+
+2. **Discover trends** in the data that were previously obscured
+
+3. **Avoid making inaccurate conclusions** about the relationships in our data
+
+To understand these benefits, let's consider Simpson's Paradox. The paradox was not invented by Homer Simpson et al., but rather Edward H. Simpson, a British codebreaker. 
+
+Simpson reported a curious phenomenon where a trend present in several different groups of data disappears when they are combined. 
+
+Below is an illustration of the Simpson's Paradox at play. Notice how the overall dataset does not have a clear direction but when you color by group, both groups exhibit the same negative trend. 
+
+::: {.panel-tabset}
+
+### Overall
+
+```{python}
+simpsons_df = pd.read_csv("data/simpsons_paradox_data_l.csv")
+alt.Chart(simpsons_df).mark_circle().encode(
+  x=alt.X('X'),
+  y=alt.Y('Y'),
+)
+```
+
+### By group
+
+```{python}
+alt.Chart(simpsons_df).mark_circle().encode(
+  color='Group',
+  x=alt.X('X'),
+  y=alt.Y('Y'),
+)
+```
+
+:::
+
+In this case, grouping makes the insidious Simpson's Paradox visually unmistakable. This is a prime example of the dual purpose of data visualization. It doesn't just present our data to others; it deepens our understanding of it. 
+
+## Strategies for grouping
+
+The grammar of graphics gives us a convenient syntax for grouping. We simply define the encoding channel that should represent a categorical variable, using `aes()` in ggplot and `.encode()` in Altair. Color, shape, size and position are the most effective channels for grouping, and they can be easily interchanged:
```

Comment:
Let's try to keep all mentions of library specific (such as `aes` and `encode`) inside the panels of the respective package